tests/unit/test_charm.py

```diff
@@ -316,20 +317,14 @@ def test_given_not_leader_and_init_secret_not_set_when_configure_then_status_is_
             WaitingStatus("Waiting for initialization secret to be set in peer relation"),
         )
 
-    @patch("charms.vault_k8s.v0.vault_client.Vault.enable_audit_device", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.is_active", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.audit_device_enabled", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.unseal", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.is_sealed", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.is_initialized", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.is_api_available", new=Mock)
-    @patch("charms.vault_k8s.v0.vault_client.Vault.is_raft_cluster_healthy", new=Mock)
+    @patch("charm.Vault", autospec=True)
```

Comment:
I like the fact that this will prevent us from forgetting to mock any aspect of the class and end up making calls to a non existent Vault. It also makes it that each test can only focus on the aspects it cares about.